### PR TITLE
Chore/add extra monitoring to gas bumping

### PR DESCRIPTION
**Pull request by samsondav**
See: https://www.pivotaltracker.com/story/show/171861047

### CODE REVIEW

**Comment by j16r** [comment]
Seems to also validate some new config value that's not getting used?

**Comment by samsondav** [comment]
@j16r Yeah the config validation was meant to get into a previous PR and I left it out by mistake so bundled it into this one.

The config value _is_ being used, it's a recently added one. I wanted to make sure we warn the user in case they set it to something silly.

File: core/store/tx_manager.go [hunk 1]
```diff
@@ -750,23 +783,27 @@ func (txm *EthTxManager) bumpGas(tx *models.Tx, attemptIndex int, blockHeight ui
 	bumpedGasPrice := txm.BumpGasByIncrement(originalGasPrice)
 
 	for {
+		promNumGasBumps.WithLabelValues(tx.Hash.String(), string(attemptIndex))
```
Comment (samsondav) [inline core/store/tx_manager.go]: Do I need an `Observe` or `Set` here? @jleeh @HenryNguyen5

Comment (jleeh) [reply core/store/tx_manager.go]: With counters @samsondav it's `Inc` or `Dec`.

File: core/store/tx_manager.go [hunk 2]
```diff
@@ -39,8 +41,38 @@ const (
...
+	)
+
+	promGasBumpUnderpricedReplacement = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "tx_manager_gas_bump_underpriced_replacement",
+		Help: "Number of underpriced replacement errors received while trying to bump gas. Counts of this type most likely indicate some kind of misconfiguration or problem.",
+	},
+		[]string{"tx_hash", "attempt_index"},
+	)
+
+	promTxAttemptFailed = promauto.NewCounterVec(prometheus.CounterOpts{
```
Comment (samsondav) [inline core/store/tx_manager.go]: Is CounterVec the right one to use?

**Comment by felder-cl** [comment]
Were these new metrics ok‘d by the team?

**Comment by samsondav** [comment]
@felder-cl What do you mean?

File: core/store/tx_manager.go (hunk 2 above)
Comment (j16r) [reply core/store/tx_manager.go]: I think you just want counter.

Comment (samsondav) [reply core/store/tx_manager.go]: I want to include the tx_hash and attempt_index though...

**Pull request merged**